src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,42 @@
+import FocusTrap from 'focus-trap-react';
+import React from 'react';
+import { Portal } from 'react-portal';
+import './Dialog.scss';
+
+export interface DialogProps {
+  children: any;
+  title: string;
+  open: boolean;
+  onClose: () => void;
+}
+
+function Dialog({children, title, open, onClose}: DialogProps) {
+  const handleBackdropClick = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.stopPropagation();
+    onClose();
+  }
+
+  if (!open) return null;
+
+  return (
+      <Portal>
+        <FocusTrap>
+          <div className="dialog" tabIndex={-1} role="dialog" onClick={handleBackdropClick} data-testid="dialog-overlay">
+            <div className="dialog__content" onClick={e => e.stopPropagation()} data-testid="dialog-content">
```

Comment:
1. To close a dialog on backdrop click, I added onClick={onClose} on the backdrop element
2. To prevent closing dialog on dialog content itself, I added onClick={e => e.stopPropagation()} to dialog content
3. Later I noticed, that when clicking on the backdrop which is on top of another clickable element (movie tile), the click will propagate to that element as well (and movie's tile handler onClick handler is called)